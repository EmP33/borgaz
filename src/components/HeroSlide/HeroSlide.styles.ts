import styled from 'styled-components';

export const ImageWrapper = styled.div`
  position: relative;
  display: flex;
  align-items: center;

  @media screen and (max-width: 600px) {
    grid-row: 1/2;
  }

  & > img {
    width: 100%;
    height: 50%;
  }

  &::before {
    content: '';
    width: 50%;
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
    background: linear-gradient(90deg, #fff 0%, transparent 100%);
    z-index: 2;

    @media screen and (max-width: 1200px) {
      width: 50%;
      background: linear-gradient(90deg, #fff 0%, transparent 100%);
    }

    @media screen and (max-width: 600px) {
      width: 100%;
      background: rgba(255, 255, 255, 0.15);
    }
  }
`;

export const ContentWrapper = styled.div`
  width: 75%;
  z-index: 3;
  padding: 1rem;

  @media screen and (max-width: 1200px) {
    width: 100%;
  }

  @media screen and (max-width: 600px) {
    grid-row: 2/3;
  }
  & p {
    text-transform: uppercase;
    color: var(--color-secondary);
  }
  & h1 {
    font-size: 5rem;
    line-height: 5rem;
    margin-top: 0.5rem;

    @media screen and (max-width: 600px) {
      font-size: 3rem;
      line-height: 3rem;
    }
  }
  & button {
    margin-top: 2rem;
  }
`;
