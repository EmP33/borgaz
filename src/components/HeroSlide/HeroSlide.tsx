import React from 'react';
import { ContentWrapper, ImageWrapper } from './HeroSlide.styles';

import { PrimaryButton } from '@/styles/Buttons.styles';
import { ISlide } from '../../pages/index';
import { GatsbyImage, getImage, IGatsbyImageData } from 'gatsby-plugin-image';

interface Props {
  slide: ISlide;
  image: IGatsbyImageData | null;
}

const HeroSlide: React.FC<Props> = ({ slide, image }) => {
  return (
    <>
      <ContentWrapper>
        <p>{slide.subheading}</p>
        <h1>{slide.heading}</h1>
        <PrimaryButton>Czytaj wiÄ™cej</PrimaryButton>
      </ContentWrapper>
      <ImageWrapper>
        {image !== undefined ? (
          <GatsbyImage image={getImage(image)} alt="basic alt" />
        ) : (
          ``
        )}
      </ImageWrapper>
    </>
  );
};

export default HeroSlide;
