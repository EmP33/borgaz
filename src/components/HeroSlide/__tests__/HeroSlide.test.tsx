import HeroSlide from '../HeroSlide';
import renderer from 'react-test-renderer';

interface ISlide {
  heading: string;
  subheading: string;
  link: string;
  id: number;
  image: number;
}
const slide: ISlide = {
  id: 0,
  heading: `STAG Q-generation`,
  subheading: `Instalacje gazowe najnowszej generacji!`,
  link: `/instalacje-gazowe`,
  image: 1,
};

describe(`<HeroSlide />`, () => {
  it(`should render properly`, async () => {
    const Hero = renderer
      .create(<HeroSlide slide={slide} image={null} />)
      .toJSON();
    expect(Hero).toMatchSnapshot();
  });
});
