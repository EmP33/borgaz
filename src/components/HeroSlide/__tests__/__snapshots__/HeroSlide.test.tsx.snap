// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroSlide /> should render properly 1`] = `
[
  <div
    className="sc-gsnTZi kxqsKk"
  >
    <p>
      Instalacje gazowe najnowszej generacji!
    </p>
    <h1>
      STAG Q-generation
    </h1>
    <button
      className="sc-dkzDqf ctKlxy"
    >
      Czytaj wiÄ™cej
    </button>
  </div>,
  <div
    className="sc-bczRLJ floyyb"
  />,
]
`;
